lib/settings/settings_screen.dart

```diff
@@ -39,37 +47,195 @@ class SettingsScreen extends StatefulWidget {
 
 class SettingsScreenState extends State<SettingsScreen> {
   int _selectedIndex = 0;
+  OrionConfig config = OrionConfig();
+  Logger logger = Logger('Settings');
+  late bool needsRestart;
+  final GlobalKey<WifiScreenState> _wifiScreenKey =
+      GlobalKey<WifiScreenState>();
+  final ValueNotifier<bool> isConnected = ValueNotifier(false);
+  late Future<void> _wifiScreenFuture;
+
+  @override
+  void initState() {
+    super.initState();
+    needsRestart = config.getFlag('needsRestart', category: 'internal');
+    _wifiScreenFuture = _initializeWifiScreen();
+  }
+
+  Future<void> _initializeWifiScreen() async {
+    final bool initialConnectionStatus = await _checkInitialConnectionStatus();
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      if (mounted) {
+        isConnected.value = initialConnectionStatus;
+      }
+    });
+  }
+
+  Future<bool> _checkInitialConnectionStatus() async {
+    try {
+      final result = await Process.run('ping', ['-c', '1', 'google.com']);
+      return result.exitCode == 0;
+    } catch (e) {
+      logger.severe('Failed to check initial connection status: $e');
+      return false;
+    }
+  }
 
   void _onItemTapped(int index) {
     setState(() {
       _selectedIndex = index;
     });
   }
```

Comment:
Agree. Maybe this is a better method?
```suggestion
  Future<bool> _checkInitialConnectionStatus() async {
    try {
      final interfaces = await NetworkInterface.list();
      return interfaces.isNotEmpty;
    } catch (e) {
      logger.severe('Failed to check initial connection status: $e');
      return false;
    }
  }
```

Reply:
Does that only list connected interfaces?

Reply:
Hmm yeah, maybe this solution is a better fit as a local replacement?
```suggestion
  Future<bool> _checkInitialConnectionStatus() async {
    try {
      // Get the default gateway IP address
      final result = await Process.run('ip', ['route', 'show', 'default']);
      if (result.exitCode != 0) {
        logger.severe('Failed to get default gateway: ${result.stderr}');
        return false;
      }

      // Extract the gateway IP address from the command output
      final gateway = RegExp(r'default via (\S+)')
          .firstMatch(result.stdout)
          ?.group(1);
      if (gateway == null) {
        logger.severe('No default gateway found');
```

Reply:
That seems good. Have you tested it on a pi and another device?

Reply:
Tested on Pi, seems to work fine! On other OSes WiFi is not supported, so any exception should be expected. Adding a check to only execute this check on Linux and simply return false on any other OS.